### PR TITLE
Bump block dependency

### PR DESCRIPTION
**Pull request by core23**
## Subject

I am targeting this branch, because this is BC.

## Changelog

```markdown
### Changed
- Bump block dependency
```

### CODE REVIEW

File: src/Block/Breadcrumb/BaseBreadcrumbMenuBlockService.php [hunk 1]
```diff
@@ -74,7 +74,7 @@ public function getName()
     /**
      * {@inheritdoc}
      */
-    public function configureSettings(OptionsResolver $resolver)
+    public function configureSettings(OptionsResolver $resolver): void
```
Comment (core23) [inline src/Block/Breadcrumb/BaseBreadcrumbMenuBlockService.php:77]: Technically this is a BC break, but if you are using sonata block bundle 4, you MUST use this signature.

File: composer.json [hunk 2]
```diff
@@ -34,14 +34,14 @@
...
-        "sonata-project/block-bundle": "<3.11",
+        "sonata-project/block-bundle": "<4.4",
         "sonata-project/core-bundle": "<3.20"
     },
     "require-dev": {
         "guzzlehttp/guzzle": "^6.0",
         "nyholm/psr7": "^1.2",
         "sonata-project/admin-bundle": "^3.31",
-        "sonata-project/block-bundle": "^3.17",
+        "sonata-project/block-bundle": "^4.4",
```
Comment (jordisala1991) [inline composer.json:44]: The other bundles are allowing 3 or 4

Comment (core23) [reply composer.json:44]: We can't do this, because we would risk a BC break in `BaseBreadcrumbMenuBlockService::configureSettings`

**Review by VincentLanglet — DISMISSED** [review]

File: composer.json (hunk 2 above)
Comment (VincentLanglet) [reply composer.json:44]: Without BlockBundle 3, the test can't run on php 7.2.

Is there a reason for dropping 7.2 support on BlockBundle 4 ?

Comment (jordisala1991) [reply composer.json:44]: All new major releases are dropping 7.2

IMO we could study what is the impact of the BC break if supporting sonata block 3, and if it is not too big, just accept it.

**Comment by core23** [comment]
Doing this on the stable branch will result in many BC breaks :(

**Pull request closed without merging**